### PR TITLE
Unit 9

### PR DESCRIPTION
**Pull request by Pavlobuch**

### CODE REVIEW

File: tests/mytests.spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const { test, expect } = require('@playwright/test');
+const exp = require('constants');
+
+async function login (page) {
```
Comment (ovartem) [inline tests/mytests.spec.js]: Well done!
There is a mechanism so called hooks - which can be used to do necessary preconditions before each test - [beforeEach](https://playwright.dev/docs/api/class-test#test-before-each-1). check out - there some others as well

File: tests/mytests.spec.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    expect(page.url()).toBe('https://www.saucedemo.com/inventory.html');
+
+    const isVisibleTitle = await page.isVisible('.title');
+    expect(isVisibleTitle).toBe(true);
+
+    const isVisibleIcon = await page.isVisible('.shopping_cart_container');
+    expect(isVisibleTitle).toBe(true);
+
+    const numberOfProducts = (await page.$$('.inventory_item')).length
+    expect(numberOfProducts).toBeGreaterThan(1)
```
Comment (ovartem) [inline tests/mytests.spec.js:25]: Great!

**Review by ovartem — CHANGES_REQUESTED** [review]
Well done!
The tests itself looks great!

But

Every JS repo should contain some additional files:
- package.json
- package-lock.json
and in your case - playwright.config.js

Without those files I won't be able to clone it and run your tests

File: playwright.config.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
+  /* Opt out of parallel tests on CI. */
+  workers: process.env.CI ? 1 : undefined,
+  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
+  reporter: 'html',
+  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+  use: {
+    /* Base URL to use in actions like `await page.goto('/')`. */
+    // baseURL: 'http://127.0.0.1:3000',
```
Comment (ovartem) [inline playwright.config.js:28]: You can add url to visit here and don't add it in every test

**Review by ovartem — CHANGES_REQUESTED** [review]
Small change is still needed

**Review by hlazkov — APPROVED** [review]
Looks nice!

**Pull request merged**